### ISSUE TITLE
Is it possible to change the style of an edge based on the node type?

### ISSUE THREAD

**Issue by guido5544** [awslabs/diagram-maker#16]
Great library!

One thing I wasn't able to figure out is how to change the style of individual edges (e.g. dashed/different color/etc). Can you let me know how this can be accomplished?

Thanks,
Guido

**Comment by sameergoyal** [comment]
It is possible to override styles for all edges globally using the following CSS overrides:

```
<containerDivId or containerDivClass> .dm-edge .dm-path.dm-path-inner {
    stroke-dasharray: 4;
    stroke: black;
}
```

The container div is added to the CSS rule to ensure the rule added by you has higher specificity than the CSS within the library itself. It is also possible to update specific edges only using the edgeId like so:

```
<containerDivId or containerDivClass> .dm-edge[data-id="edgeId"] .dm-path.dm-path-inner {
    stroke-dasharray: 4;
    stroke: black;
}
```

However, there is no information present on the rendered edge around the node types itself that can be used for style overrides. Can you describe your use case further? If we were to add data attributes or CSS classes on the edge for the sourceNodeType & destinationNodeType, would you be able to use CSS overrides using those to satisfy your use case/requirements?

**Comment by sameergoyal** [comment]
We have pulled the change to add this information in the form of data attributes to the edges. You would be able to use it to add CSS overrides on edges based on source & destination node types.

This will be included with our next release.

**Comment by sameergoyal** [comment]
We have released a new version: [1.1.0](https://www.npmjs.com/package/diagram-maker) that contains this feature.

Closing the issue.
